# **Exam 2022**

First we read the Data.

Tip: To insert a newline without sending your message, try the following:

- Shift+Enter (works in many chat input boxes).
- If Shift+Enter doesn't work, try Ctrl+Enter or Alt+Enter.
- Check the chat input settings (gear/menu) to switch between "Enter sends" and "Enter inserts newline".

In [2]:
breakfast <- data.frame("name"= c("100Bran","100NaturalBran","AllBran","AllBranFiber","AlmondDelight","AppleCinnamonCheerios","AppleJacks","Basic4","BranChex","BranFlakes","CapCrunch","Cheerios","CinnamonToastCrunch","Clusters","CocoaPuffs","CornChex","CornFlakes","CornPops","CountChocula","CracklinOatBran","CreamWheat","Crispix","CrispyWheatRaisins","DoubleChex","FrootLoops","FrostedFlakes","FrostedMiniWheats","FruitFibreDates","FruitfulBran","FruityPebbles","GoldenCrisp","GoldenGrahams","GrapeNutsFlakes","GrapeNuts","GreatGrainsPecan","HoneyGraham","HoneyNutCheerios","Honeycomb","JustRightCrunchyNuggets","JustRightFruitNut","Kix","Life","LuckyCharms","MuesliRaisinsDatesAlmonds","MuesliRaisinsPeachesPecans","MueslixCrispyBlend","MultiGrainCheerios","NutHoneyCrunch","NutriGrainAlmondRaisin","NutrigrainWheat","OatmealRaisinCrisp","PostNatRaisinBran","Product19","PuffedRice","PuffedWheat","QuakerOatSquares","QuakerOatmeal","RaisinBran","RaisinNutBran","RaisinSquares","RiceChex","RiceKrispies","ShreddedWheat","ShreddedWheatBran","ShreddedWheatspoonsize","Smacks","SpecialK","StrawberryFruitWheats","TotalCornFlakes","TotalRaisinBran","TotalWholeGrain","Triples","Trix","WheatChex","Wheaties","WheatiesHoneyGold"),
                        "mfr" = factor(c(3,5,2,2,6,1,2,1,6,4,5,1,1,1,1,6,2,2,1,2,3,2,1,6,2,2,2,4,2,4,4,1,4,4,4,5,1,4,2,2,1,5,1,6,6,2,1,2,2,2,1,4,2,5,5,5,5,2,1,2,6,2,3,3,3,2,2,3,1,1,1,1,1,6,1,1), levels= c(1,2,3,4,5,6), labels = c("G", "K", "N", "P", "Q", "R")),
                        "type" = factor(c("C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","H","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","H","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C","C")),
                        "calories" = c(70,120,70,50,110,110,110,130,90,90,120,110,120,110,110,110,100,110,110,110,100,110,100,100,110,110,100,120,120,110,100,110,100,110,120,120,110,110,110,140,110,100,110,150,150,160,100,120,140,90,130,120,100,50,50,100,100,120,100,90,110,110,80,90,90,110,110,90,110,140,100,110,110,100,100,110),
                        "protein" = c(4,3,4,4,2,2,2,3,2,3,1,6,1,3,1,2,2,1,1,3,3,2,2,2,2,1,3,3,3,1,2,1,3,3,3,1,3,1,2,3,2,4,2,4,4,3,2,2,3,3,3,3,3,1,2,4,5,3,3,2,1,2,2,3,3,2,6,2,2,3,3,2,1,3,3,2),
                        "fat" = c(1,5,1,0,2,2,0,2,1,0,2,2,3,2,1,0,0,0,1,3,0,0,1,0,1,0,0,2,0,1,0,1,1,0,3,2,1,0,1,1,1,2,1,3,3,2,1,1,2,0,2,1,0,0,0,1,2,1,2,0,0,0,0,0,0,1,0,0,1,1,1,1,1,1,1,1),
                        "sodium" = c(130,15,260,140,200,180,125,210,200,210,220,290,210,140,180,280,290,90,180,140,80,220,140,190,125,200,0,160,240,135,45,280,140,170,75,220,250,180,170,170,260,150,180,95,150,150,220,190,220,170,170,200,320,0,0,135,0,210,140,0,240,290,0,0,0,70,230,15,200,190,200,250,140,230,200,200),
                        "fiber" = c(10,2,9,14,1,1.5,1,2,4,5,0,2,0,2,0,0,1,1,0,4,1,1,2,1,1,1,3,5,5,0,0,0,3,3,3,1,1.5,0,1,2,0,2,0,3,3,3,2,0,3,3,1.5,6,1,0,1,2,2.7,5,2.5,2,0,0,3,4,3,1,1,3,0,4,3,0,0,3,3,1),
                        "carbo" = c(5,8,7,8,14,10.5,11,18,15,13,12,17,13,13,12,22,21,13,12,10,21,21,11,18,11,14,14,12,14,13,11,15,15,17,13,12,11.5,14,17,20,21,12,12,16,16,17,15,15,21,18,13.5,11,20,13,10,14,-1,14,10.5,15,23,22,16,19,20,9,16,15,21,15,16,21,13,17,17,16),
                        "sugars" = c(6,8,5,0,8,10,14,8,6,5,12,1,9,7,13,3,2,12,13,7,0,3,10,5,13,11,7,10,12,12,15,9,5,3,4,11,10,11,6,9,3,6,12,11,11,13,6,9,7,2,10,14,3,0,0,6,-1,12,8,6,2,3,0,0,0,15,3,5,3,14,3,3,12,3,3,8),
                        "potass" = c(280,135,320,330,-1,70,30,100,125,190,35,105,45,105,55,25,35,20,65,160,-1,30,120,80,30,25,100,200,190,25,40,45,85,90,100,45,90,35,60,95,40,95,55,170,170,160,90,40,130,90,120,260,45,15,50,110,110,240,140,110,30,35,95,140,120,40,55,90,35,230,110,60,25,115,110,60),
                        "vitamins" = c(25,0,25,25,25,25,25,25,25,25,25,25,25,25,25,25,25,25,25,25,0,25,25,25,25,25,25,25,25,25,25,25,25,25,25,25,25,25,100,100,25,25,25,25,25,25,25,25,25,25,25,25,100,0,0,25,0,25,25,25,25,25,0,0,0,25,25,25,100,100,100,25,25,25,25,25),
                        "shelf" = c(3,3,3,3,3,1,2,3,1,3,2,1,2,3,2,1,1,2,2,3,2,3,3,3,2,1,2,3,3,2,1,2,3,3,3,2,1,1,3,3,2,2,2,3,3,3,1,2,3,3,3,3,3,3,3,3,1,2,3,3,1,1,1,1,1,2,1,2,3,3,3,3,2,1,1,1),
                        "weight" = c(1,1,1,1,1,1,1,1.33,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1.25,1.33,1,1,1,1,1,1,1,1,1,1,1.3,1,1,1,1,1,1.5,1,1,1.33,1,1.25,1.33,1,0.5,0.5,1,1,1.33,1,1,1,1,0.83,1,1,1,1,1,1,1.5,1,1,1,1,1,1),
                        "cups" = c(0.33,1,0.33,0.5,0.75,0.75,1,0.75,0.67,0.67,0.75,1.25,0.75,0.5,1,1,1,1,1,0.5,1,1,0.75,0.75,1,0.75,0.8,0.67,0.67,0.75,0.88,0.75,0.88,0.25,0.33,1,0.75,1.33,1,0.75,1.5,0.67,1,1,1,0.67,1,0.67,0.67,1,0.5,0.67,1,1,1,0.5,0.67,0.75,0.5,0.5,1.13,1,1,0.67,0.67,0.75,1,1,1,1,1,0.75,1,0.67,1,0.75),
                        "rating" = c(68.402973,33.983679,59.425505,93.704912,34.384843,29.509541,33.174094,37.038562,49.120253,53.313813,18.042851,50.764999,19.823573,40.400208,22.736446,41.445019,45.863324,35.782791,22.396513,40.448772,64.533816,46.895644,36.176196,44.330856,32.207582,31.435973,58.345141,40.917047,41.015492,28.025765,35.252444,23.804043,52.076897,53.371007,45.811716,21.871292,31.072217,28.742414,36.523683,36.471512,39.241114,45.328074,26.734515,37.136863,34.139765,30.313351,40.105965,29.924285,40.69232,59.642837,30.450843,37.840594,41.50354,60.756112,63.005645,49.511874,50.828392,39.259197,39.7034,55.333142,41.998933,40.560159,68.235885,74.472949,72.801787,31.230054,53.131324,59.363993,38.839746,28.592785,46.658844,39.106174,27.753301,49.787445,51.592193,36.187559))

### **Question 1.1**

<img src="SymPyBilleder/2025-11-10-10-41-30.png" width="550">

**Answer:**

In [8]:
# We have to run MANOVA:
variables <- c("calories", "protein", "fat", "sodium", "fiber",
               "carbo", "sugars", "potass", "rating")

# Create formula dynamically
formula <- as.formula(paste("cbind(", paste(variables, collapse = ", "), ") ~ mfr"))

# Run MANOVA
manova_model <- manova(formula, data = breakfast)

# Display MANOVA summary with Wilks' Lambda
summary(manova_model, test = "Wilks")

          Df  Wilks approx F num Df den Df   Pr(>F)    
mfr        5 0.1661   3.0774     45 280.44 7.46e-09 ***
Residuals 70                                           
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Here, the Wilks gives $0.1661$.

**Hence, the correct answer is option 2.**

### **Question 1.2**

<img src="SymPyBilleder/2025-11-10-10-46-11.png" width="550">

**Answer:**

In [12]:
# We need type III SS for the breakfast data:
summary.aov(manova_model)

 Response calories :
            Df  Sum Sq Mean Sq F value  Pr(>F)  
mfr          5  4661.8  932.35  2.7033 0.02724 *
Residuals   70 24142.2  344.89                  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

 Response protein :
            Df Sum Sq Mean Sq F value Pr(>F)
mfr          5  2.027 0.40534  0.3264 0.8954
Residuals   70 86.921 1.24172               

 Response fat :
            Df Sum Sq Mean Sq F value   Pr(>F)   
mfr          5 15.695 3.13909  3.5851 0.006057 **
Residuals   70 61.291 0.87559                    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

 Response sodium :
            Df Sum Sq Mean Sq F value    Pr(>F)    
mfr          5 180643   36129  7.7188 7.925e-06 ***
Residuals   70 327643    4681                      
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

 Response fiber :
            Df Sum Sq Mean Sq F value  Pr(>F)  
mfr          5  54.81 10.9625  2.0617 0.08054 .
Residuals   70 37

The output gives us F valuesand a corresponding SS value for how much manufacturers differ on that variable. From all the values, we see that sodium has the highest value of $180643$.

**Therefore, the correct answer is option 4.**

### **Question 1.3**

<img src="SymPyBilleder/2025-11-10-10-50-51.png" width="550">

**Answer:**

In [22]:
library(stats)

# Get Wilks lambda from model
lambda <- summary(manova_model, test="Wilks")$stats[1, 2]
# Compute U approx parameters
p <- 9
g <- 6
N <- nrow(breakfast)

U_params <- c(p, g-1, N - g)
U_params

[1]  9  5 70

So we see that this matches option 3 WITH EXPLANATION:

Good — this question is about the **distribution of the test statistic under the null hypothesis** in a **MANOVA** (multivariate ANOVA).

We test this first when we recall the model setup:
$$
H_0: \mu_G = \mu_K = \mu_N = \mu_P = \mu_Q = \mu_R
$$

Then against the general alternative. That is, we have $p = 9$ response variables (`calories, protein, fat, sodium, fiber, carbo, sugars, potass, rating`), $g = 6$ groups (manufacturers) and $g – 1 = 5$ numerator groups for MANOVA.

Furthermore, the Wilks’ Lambda statistic is:
$$
\Lambda = \frac{|\mathbf{E}|}{|\mathbf{E} + \mathbf{H}|}
$$

And its corresponding test statistic ( $U$ ) (Rao’s U transformation) follows approximately a U-distribution* with parameters depending on:

> $p =$ number of dependent variables (9)\
> $g =$ number of groups (6)\
> $N =$ total sample size (number of cereals $= 77$ in this dataset)

The transformed statistic:
$$
U = \frac{(1 - \Lambda^{1/s})}{\Lambda^{1/s}}
$$
Approximately follows:
$$
U(p, g-1, N - g)
$$
Where $s = \sqrt{\frac{p^2(g-1)^2 - 4}{p^2 + (g-1)^2 - 5}} $.

THEN we plug in our values $p = 9$, $g - 1 = 5$ and  $N - g \approx 77 - 6 = 71$ with rounded values often used in the question setup correspond to $U(9, 5, 70)$.

**Therefore, the answer is option 3.**

### **Question 1.4**

<img src="SymPyBilleder/2025-11-10-11-05-56.png" width="550">

**Answer:**

In [27]:
# Given parameters
u <- 0.2        # observed test statistic
s <- 1          # first parameter in U(s, r, n-k)
r <- 2          # second parameter
k <- 2          # model parameter (from U(1,2,n-2))
alpha <- 0.05   # significance level
n_values <- 5:9 # n range to check

# Initialize vectors to store results
F_statistics <- numeric(length(n_values))
p_values <- numeric(length(n_values))

# Loop over possible sample sizes
for (i in seq_along(n_values)) {
  n <- n_values[i]
  # Compute v (adjustment factor)
  v <- 0.5 * (2 * (n - k) + r - s - 1)
  # Compute t (correction exponent)
  t <- ifelse(s^2 + r^2 == 5, 
              1, 
              sqrt((s^2 * r^2 - 4) / (s^2 + r^2 - 5)))
  # Compute the transformed F-statistic
  F_statistic <- ((1 - u^(1/t)) / (u^(1/t))) * ((v * t + 1 - 0.5 * s * r) / (s * r))
  # Degrees of freedom for the F distribution
  df1 <- s * r
  df2 <- v * t + 1 - 0.5 * s * r
  # p-value
  p_value <- 1 - pf(F_statistic, df1, df2)
  # Store results
  F_statistics[i] <- F_statistic
  p_values[i] <- p_value
}

# Combine into a data frame
results <- data.frame(n = n_values, F_statistic = round(F_statistics, 4), p_value = round(p_values, 4), significant = p_values < alpha)

print(results)

  n F_statistic p_value significant
1 5           6  0.0894       FALSE
2 6           8  0.0400        TRUE
3 7          10  0.0179        TRUE
4 8          12  0.0080        TRUE
5 9          14  0.0036        TRUE


We see that it is already significant at $n=6$.

**Therefore, the correct answer is option 3.** (CHAT GOT IT WRONG)

### **Question 2.1**

We still consider the data introduced in Problem 1.

We now consider whether we can discriminate between the different manufacturers by means of a Linear Discriminant Analysis.
We only consider the following variables *calories protein fat sodium fiber carbo sugars potass.*

<img src="SymPyBilleder/2025-11-10-11-15-42.png" width="550">

**Answer:**

In [6]:
# # Load dataset
# source("./exam_data/breakfastData.R")

# Select only relevant variables
vars <- c("calories", "protein", "fat", "sodium", "fiber", "carbo", "sugars", "potass")

# Perform Linear Discriminant Analysis
library(MASS)
lda_model <- lda(mfr ~ ., data = breakfast[, c("mfr", vars)], prior = rep(1/6, 6))

# Predict (resubstitution)
lda_pred <- predict(lda_model)

# Confusion matrix
table(Predicted = lda_pred$class, Actual = breakfast$mfr)

# Count misclassifications
misclassified <- sum(lda_pred$class != breakfast$mfr)
cat("Number of misclassifications:", misclassified, "\n")

         Actual
Predicted  G  K  N  P  Q  R
        G 14  2  0  0  3  1
        K  0  9  0  2  0  0
        N  0  3  5  0  0  0
        P  2  3  1  6  0  0
        Q  1  1  0  0  5  0
        R  5  5  0  1  0  7

Number of misclassifications: 30 


Here we see that the output gives $30$ misclassified observations.

**Therefore, the answer option 4.**

### **Question 2.2**

<img src="SymPyBilleder/2025-11-10-11-25-23.png" width="550">

**Answer:**

In [22]:
library(MASS)

lda_fit <- lda(mfr ~ calories + protein + fat + sodium + fiber + carbo + sugars + potass,
               data = breakfast)

# Mahalanobis distances between class means
centroids <- lda_fit$means
S_pooled <- lda_fit$scaling %*% t(lda_fit$scaling)
mahal <- as.matrix(dist(scale(centroids), method = "euclidean"))
mahal

G        K        N        P        Q        R       
G 0.000000 2.901146 6.394279 2.652049 3.624195 1.817181
K 2.901146 0.000000 4.227419 1.702450 4.035261 2.226869
N 6.394279 4.227419 0.000000 4.841469 5.540727 5.511853
P 2.652049 1.702450 4.841469 0.000000 3.984151 2.836417
Q 3.624195 4.035261 5.540727 3.984151 0.000000 4.095831
R 1.817181 2.226869 5.511853 2.836417 4.095831 0.000000

HERE, we have to note that a small Mahalanobis distance means the two groups (manufacturers) are hard to discriminate, for instance, cereals from those manufacturers look very similar in terms of nutrients.

We see that out of the combination options, then $K$ and $P$ have lowest value out of all.

**Therefore, the correct answer is option 1.**

### **Question 2.3**

<img src="SymPyBilleder/2025-11-10-11-35-11.png" width="550">

**Answer:**

In [36]:
# Load required package
library(Hotelling)

# Subset data for manufacturers K and R
breakfast_KR <- subset(breakfast, mfr %in% c("K", "R"))

# Select only the relevant variables
vars <- c("calories", "protein", "fat", "sodium", "fiber", "carbo", "sugars", "potass")

# Perform Hotelling's T² test
hotelling_result <- hotelling.test(
  breakfast_KR[breakfast_KR$mfr == "K", vars],
  breakfast_KR[breakfast_KR$mfr == "R", vars]
)

# View result
hotelling_result

Test stat:  12.616 
Numerator df:  8 
Denominator df:  22 
P-value:  0.3456 

(CHAT GOT IT WRONG TO BEGIN WITH).

### **Question 2.4**

<img src="SymPyBilleder/2025-11-10-12-24-03.png" width="550">

**Answer:**

In [43]:
# Subset to K and R
kr <- subset(breakfast, mfr %in% c("K", "R"))

# Make sure the data are clean (remove NAs)
kr <- na.omit(kr)

# Variables to test
vars <- c("protein", "potass")

# Create the two groups
K <- kr[kr$mfr == "K", vars]
R <- kr[kr$mfr == "R", vars]

# Perform Hotelling's T2 test
res_hot <- hotelling.test(K, R, perm = FALSE)

print(res_hot)

Test stat:  0.1734 
Numerator df:  2 
Denominator df:  28 
P-value:  0.9199 


(CHAT KEEPS GETTING IT WRONG).

**The correct answer should be option 3 with $[0.6, 0.8[$.**

<img src="SymPyBilleder/2025-11-10-12-34-38.png" width="550">

### **Question 3.1**

<img src="SymPyBilleder/2025-11-10-12-34-54.png" width="550">

**Answer:**

OUR reasoning is that these variables are measured in very different units and scales:

> Calories is tens to hundreds\
> protein is grams (small numbers, around 1–6)\
> sodium is milligrams (hundreds)\
> potass is milligrams (hundreds)

When variable scales differ drastically, the covariance matrix will cause the variables with the largest variances (like sodium or calories) to dominate the principal components. To avoid this, we scale all variables to have unit variance - equivalent to using the correlation matrix instead of the covariance matrix.

Thus, “The correlation matrix, since the variables are on a very different scale.” is correct. BUT why not the others?

> Covariance matrix is wrong, because the scale differences distort results.\
> Correlation matrix, as it compresses the data is not the main reason.\
> PCA automatically rescales the data is false, it does not because you must standardize manually or use `cor = TRUE` in R’s `prcomp()`.\
> Covariance matrix since data are on a different scale is completely wrong reasoning.

Example in R:

```r
vars <- c("calories", "protein", "fat", "sodium", "fiber", "carbo", "sugars", "potass")
pca_res <- prcomp(breakfast[, vars], scale. = TRUE)  # uses correlation matrix
summary(pca_res)
```

**Therefore, the correct answer is option 3.**

### **Question 3.2**

<img src="SymPyBilleder/2025-11-10-12-49-31.png" width="550">

**Answer:**

We take the last two eigenvalues so $7$ and $8$, and we have to use the following formula, ebcause we are considering the correlation matrix:
$$
Z_2 = -n \log \frac{\text{det}\hat{R}}{\hat{\lambda_1} \cdot ... \cdot \hat{\lambda}_m\cdot \hat{\lambda}^{k-m}} 
$$
$$
-n \log \frac{\hat{\lambda}_{m+1} \cdot ... \cdot \hat{\lambda}_k}{\hat{\lambda}_*^{k-m}}
$$
Where:
$$
\hat{\lambda}_* = (\hat{\lambda}_{m+1} + \dots + \hat{\lambda}_{k}) / (k - m)
$$
Here, $n$ is number of observations, $k$ is the number of variables being analyzed (the dimension of the correlation matrix), $m$ is the number of factors we are retaining before testing the equality of remaining eigenvalues.

We just insert everything in R.

In [46]:
k <- 8 # we have 8 eigenvalues
n <- 76
m <- 6 # eigenvalues after removing the two we want to analyze

e7 <- 0.06890717
e8 <- 0.05505174

hatlambdastar <- (e7 + e8) / (k - m)

Z2 <- -n * log((e7 * e8) / (hatlambdastar)^(k - m))
Z2

[1] 0.9554881

**The answer is, therefore, option 1.**

### **Question 3.3**

<img src="SymPyBilleder/2025-11-12-16-25-34.png" width="550">

<img src="SymPyBilleder/2025-11-12-16-25-50.png" width="550">

**Answer:**

First option is wrong, because factor 1 is mainly a weigthing for potass and fiber, meanwhile factor 3 is for sodium and carbo.

Second option is false, because it states that factor 3 weights fat high, but fat is at the bottom for this factor.

Thrid option is correct for now.

Fourth option is false, it is not an average of all variables.

Fifth option is false, because it states that factor 3 is mainly a contrast between sugars versus carbo and fat, but carbo is pretty high up meanwhile fat is low.

**Therefore, the correct answer is option 3.**

### **Question 3.4**

<img src="SymPyBilleder/2025-11-12-16-30-43.png" width="550">

**Answer:**

The uniqueness of a variable in factor analysis is given by the formula:
$$
i = 1- \sum_{j=1}^{m} a_{ij}^2
$$
So we see that it must be the last option.

**The corect answer is, therefore, option 6.**

### **Question 3.5**

<img src="SymPyBilleder/2025-11-12-16-41-12.png" width="550">

**Answer:**

We use this formula for communality:
$$
h_i^2 = \sum_{j=1}^{m} a_{ij}^2
$$

In [53]:
(0.71036)^2 + (0.40915)^2

[1] 0.6720151

**The answer is, thus, option 1.**

### **Questionn 4.1**

<img src="SymPyBilleder/2025-11-12-16-55-45.png" width="550">

<img src="SymPyBilleder/2025-11-12-16-56-16.png" width="550">

**Answer:**

The question directly refers to how much of the variation in the dependent variable (rating) is explained by the fitted model. We have to use the formula:
$$
R^2 = 1 - \frac{SS_{res}}{SS_{tot}}
$$

In [59]:
# We first load our breakfast data and then make the model:
b_model <- lm(rating ~ calories + protein + fat + sodium + fiber + carbo + sugars + potass, data = breakfast)

SS_res <- sum(residuals(b_model)^2)
SS_tot <- sum((breakfast$rating - mean(breakfast$rating))^2)

SS_res
SS_tot

[1] 79.97979

[1] 14846.37

The numbers fit option 1.

**The answer is, therefore, option 1.**

### **Question 4.2**

<img src="SymPyBilleder/2025-11-12-17-05-02.png" width="550">

**Answer:**

In [60]:
summary(b_model)


Call:
lm(formula = rating ~ calories + protein + fat + sodium + fiber + 
    carbo + sugars + potass, data = breakfast)

Residuals:
    Min      1Q  Median      3Q     Max 
-3.6173 -0.0386  0.2240  0.5709  1.2172 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) 55.989429   0.915760  61.140  < 2e-16 ***
calories    -0.228688   0.015988 -14.304  < 2e-16 ***
protein      3.207625   0.184747  17.362  < 2e-16 ***
fat         -1.601468   0.200146  -8.002 2.39e-11 ***
sodium      -0.058024   0.001724 -33.654  < 2e-16 ***
fiber        3.415056   0.158390  21.561  < 2e-16 ***
carbo        1.036456   0.060776  17.054  < 2e-16 ***
sugars      -0.767629   0.064515 -11.898  < 2e-16 ***
potass      -0.034522   0.005232  -6.599 7.87e-09 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1.093 on 67 degrees of freedom
Multiple R-squared:  0.9946,	Adjusted R-squared:  0.994 
F-statistic:  1546 on 8 and 67 DF,  p-value: < 2.

For the fraction of variation to be explained by the model is $0$, then we must have the hypothesis that $H_0: \beta_1 = \beta_2 = ... = \beta_8 = 0$, which means that the model does not explain any of the variation in the dependent variable (rating). This is tested using an F-test aka:
$$
F(k, n-k-1)
$$
We find the correct parameters in R.

In [61]:
k <- 8 # number of predictors no intercept
n <- nrow(breakfast)

n-k-1

[1] 67

So we must have like $F(8, 67)$.

**The answer is, therefore, option 3.**

### **Question 4.3**

<img src="SymPyBilleder/2025-11-12-17-09-58.png" width="550">

**Answer:**

Firstly, from the book? It is known that the rule of thumb is $TOL < 0.1$ or $VIF > 10$ indicates multicollinearity problem. First we have to find all values, we do that in R.

In [64]:
# Finding TOL values in R:
library(car)
vif_values <- vif(b_model)
tol_values <- 1 / vif_values

vif_values
tol_values 

calories  protein      fat   sodium    fiber    carbo   sugars   potass 
6.167919 2.543206 2.583475 1.265749 8.974267 4.299526 5.181118 8.855874

calories   protein       fat    sodium     fiber     carbo    sugars    potass 
0.1621292 0.3932045 0.3870756 0.7900461 0.1114297 0.2325838 0.1930085 0.1129194

We see that the lost is indeed $0.1129$ and the highest VIF is $8.97$, so this matches with either option 1 or option 4. But these values are both outside the multicollinearity problem range, so we choose option 1.

**The correct answer is, therefore, option 1.**

### **Question 4.4**

<img src="SymPyBilleder/2025-11-12-19-01-48.png" width="550">

**Answer:**

In [67]:
# We print cook's distance for all variables:
cooksd <- cooks.distance(b_model)

# We find the most influential observation:
max_cook <- which.max(cooksd)
max_cook_value <- cooksd[max_cook]

max_cook
max_cook_value

70 
70

70 
0.2096493

In [68]:
# THEN for DFFITS:
dffits_values <- dffits(b_model)
max_dffits_index <- which.max(abs(dffits_values))
max_dffits_value <- dffits_values[max_dffits_index]

max_dffits_index
max_dffits_value

70 
70

70 
-1.459995

**The answer is, therefore, the last option.**

### **Question 4.5**

<img src="SymPyBilleder/2025-11-12-19-02-05.png" width="550">

**Answer:**

We can write code in R for this, but generally to check for observations with the largest potential to influence estimates, we can look at hat values $h_{ii}$, Cook's distance, and DFFITS and DFBETAS.

In [71]:
which.max(hatvalues(b_model))
which.max(cooks.distance(b_model))
which.max(dffits(b_model))
which.max(dfbetas(b_model))

57 
57

70 
70

21 
21

[1] 513

BUT, since the question specifically asks for POTENITAL (measured by leverage), THEN we use hat values! If it asked for "largest influence" then it should be Cook's distance.

**The answer is, therefore, the last option with $57$.**


### **Question 4.6**

<img src="SymPyBilleder/2025-11-12-19-02-21.png" width="550">

**Answer:**

For this, we know we have $t$-values in summary output, and $t^2_j = F_{1,n-p}$ for each variable $j$.

In [77]:
s <- summary(b_model)
tvals <- s$coefficients[, "t value"]
Fvals <- tvals^2

# Find the lowest
which.min(Fvals)

potass 
     9

In [78]:
summary(b_model)


Call:
lm(formula = rating ~ calories + protein + fat + sodium + fiber + 
    carbo + sugars + potass, data = breakfast)

Residuals:
    Min      1Q  Median      3Q     Max 
-3.6173 -0.0386  0.2240  0.5709  1.2172 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) 55.989429   0.915760  61.140  < 2e-16 ***
calories    -0.228688   0.015988 -14.304  < 2e-16 ***
protein      3.207625   0.184747  17.362  < 2e-16 ***
fat         -1.601468   0.200146  -8.002 2.39e-11 ***
sodium      -0.058024   0.001724 -33.654  < 2e-16 ***
fiber        3.415056   0.158390  21.561  < 2e-16 ***
carbo        1.036456   0.060776  17.054  < 2e-16 ***
sugars      -0.767629   0.064515 -11.898  < 2e-16 ***
potass      -0.034522   0.005232  -6.599 7.87e-09 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1.093 on 67 degrees of freedom
Multiple R-squared:  0.9946,	Adjusted R-squared:  0.994 
F-statistic:  1546 on 8 and 67 DF,  p-value: < 2.

**The answer is, therefore, option 1.**

### **Question 4.7**

<img src="SymPyBilleder/2025-11-12-19-02-35.png" width="550">

**Answer:**

In [79]:
summary(b_model)


Call:
lm(formula = rating ~ calories + protein + fat + sodium + fiber + 
    carbo + sugars + potass, data = breakfast)

Residuals:
    Min      1Q  Median      3Q     Max 
-3.6173 -0.0386  0.2240  0.5709  1.2172 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) 55.989429   0.915760  61.140  < 2e-16 ***
calories    -0.228688   0.015988 -14.304  < 2e-16 ***
protein      3.207625   0.184747  17.362  < 2e-16 ***
fat         -1.601468   0.200146  -8.002 2.39e-11 ***
sodium      -0.058024   0.001724 -33.654  < 2e-16 ***
fiber        3.415056   0.158390  21.561  < 2e-16 ***
carbo        1.036456   0.060776  17.054  < 2e-16 ***
sugars      -0.767629   0.064515 -11.898  < 2e-16 ***
potass      -0.034522   0.005232  -6.599 7.87e-09 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1.093 on 67 degrees of freedom
Multiple R-squared:  0.9946,	Adjusted R-squared:  0.994 
F-statistic:  1546 on 8 and 67 DF,  p-value: < 2.

Positive increases rating while negative decreases rating.

We see that fiber and then has the highest intercept values, meanwhile calories, sodium, sugars an potass are negative. So therefore, the most agreeable is the last option, because we should prioritize the fiber, protein and carbonhydrates, meanwhile the rest should be low.

**The answer is, therefore, option 6.**

### **Question 5.1**

<img src="SymPyBilleder/2025-11-12-19-02-54.png" width="550">

<img src="SymPyBilleder/2025-11-12-19-04-46.png" width="550">

**Answer:**

First we have a model design matrix:
$$
X = \begin{bmatrix} 1 & 1 & 0 \\ 1 & -1 & 0 \\ 1 & -1 & 0 \\ 1 & 1 & 1 \end{bmatrix}
$$
Then we see the actual least squares estimate in the task description, and we see that the first line is:
$$
\frac{1}{4} \begin{bmatrix} 2& 1 & 1 & 0 \end{bmatrix}
$$
**The answer is, therefore, option 4.**

### **Question 5.2**

<img src="SymPyBilleder/2025-11-12-19-05-05.png" width="550">

**Answer:**

So, we have the formula:
$$
V(\hat{\beta}) = \sigma^2 (X^TX)^{-1}
$$
We can see from the task description that the $(X^TX)^{-1}$ is multiplied with something like $\frac{1}{8}$, also to be sure we write it manually in R.

In [83]:
X <- matrix(c(1, 1, 0,
              1, -1, 0,
              1, -1, 0,
              1, 1, 1), nrow = 4, byrow = TRUE)

solve(t(X) %*% X)

[,1]   [,2]   [,3]
[1,]  0.375  0.125 -0.5
[2,]  0.125  0.375 -0.5
[3,] -0.500 -0.500  2.0

This proves the $\frac{1}{8}$ factor, but then we have to take diagonal elements of the matrix because that is the variance, and we have $3$ as the first diagnal element so $\frac{3}{8}$ becuase our model right now is just the first row.

**The answer is, therefore, option 4.**

### **Question 5.3**

<img src="SymPyBilleder/2025-11-12-19-05-25.png" width="550">

**Answer:**

This must be another row of the resulting least squares estimate matrix, we see the only matching row is:
$$
\begin{bmatrix} -1 & 0 & 0 & 1 \end{bmatrix} = \frac{1}{4} \begin{bmatrix} -4 & 0 & 0 & 4 \end{bmatrix}
$$
**The answer is, therefore, option 3.**

### **Question 5.4**

<img src="SymPyBilleder/2025-11-12-19-05-41.png" width="550">

**Answer:**

So we need to find the variances first and then covariance. The variances are:
$$
V(\hat{\mu}) = \frac{3}{8} \sigma^2
$$
$$
V(\hat{\beta_1}) = \frac{16}{8} \sigma^2
$$
(we read from the table where model 1 is row 1 and model 2 is row 3). Then the covariance is given as:
$$
Cov(\hat{\mu}, \hat{\beta_1}) = -\frac{4}{8} \sigma^2
$$
This value is evident from the $(X^TX)^{-1}$ matrix with the element that is row 3 and column 1.

Then we can use the formula for correlation:
$$
\rho = \frac{Cov(X, Y)}{\sqrt{V(X)V(Y)}}
$$
Inserting everything gives:
$$
\rho = \frac{-\frac{4}{8} \sigma^2}{\sqrt{\frac{3}{8} \sigma^2 \cdot \frac{16}{8} \sigma^2}} = -\frac{4/8}{\sqrt{48}/8} = -\frac{4}{\sqrt{48}} = -\frac{1}{\sqrt{3}} 
$$
This is basically also $-\frac{\sqrt{3}}{3}$.

**The answer is, therefore, option 1.**

### **Question 5.5**

<img src="SymPyBilleder/2025-11-12-19-05-57.png" width="550">

**Answer:**

First we have to modify our desing matrix now since $\beta = 0$, so:
$$
X = \begin{bmatrix} 1 & 1 \\ 1 & -1 \\ 1 & -1 \\ 1 & 1 \end{bmatrix}
$$
Then we do least squares in R.

In [87]:
Xnob <- matrix(c(1, 1,
                          1, -1,
                          1, -1,
                          1, 1), nrow = 4, byrow = TRUE)

solve(t(Xnob) %*% Xnob) %*% t(Xnob)

[,1] [,2]  [,3]  [,4]
[1,] 0.25  0.25  0.25 0.25
[2,] 0.25 -0.25 -0.25 0.25

The first line matches option 2.

**The answer is, therefore, option 2.**

### **Question 5.6**

<img src="SymPyBilleder/2025-11-12-19-06-10.png" width="550">

**Answer:**

We check between the model $\frac{1}{4} \begin{bmatrix} 2 & 1 & 1 & 0 \end{bmatrix}$ and $\frac{1}{4} \begin{bmatrix} 1 & 1 & 1 & 1 \end{bmatrix}$. We use the formula:
$$
C(AX, BY) = AC(X,Y)B^T
$$
So:
$$
\frac{1}{4} \begin{bmatrix} 2 & 1 & 1 & 0 \end{bmatrix} \cdot \begin{bmatrix} \sigma^2 & 0 &0 &0 \\ 0 & \sigma^2 & 0 & 0 \\ 0 & 0 & \sigma^2 & 0 \\ 0 & 0 & 0 & \sigma^2 \end{bmatrix} \cdot \frac{1}{4} \begin{bmatrix} 1 \\ 1 \\ 1 \\ 1 \end{bmatrix} = \frac{1}{16} \begin{bmatrix} 2 & 1 & 1 & 0 \end{bmatrix} \cdot \begin{bmatrix} \sigma^2 \\ \sigma^2 \\ \sigma^2 \\ \sigma^2 \end{bmatrix} = \frac{1}{16} \cdot 4\sigma^2 = \frac{1}{4} \sigma^2
$$
Now we have the covariance, we can find correlation now.

In [88]:
solve(t(Xnob) %*% Xnob)

[,1] [,2]
[1,] 0.25 0.00
[2,] 0.00 0.25

So:
$$
\rho = \frac{\frac{1}{4} \sigma^2}{\sqrt{\frac{3}{8} \sigma^2 \cdot \frac{1}{4} \sigma^2}} = \frac{\sqrt{2}}{\sqrt{3}} = \frac{\sqrt{6}}{3}
$$
**The answer is, thus, option 4.**

### **Question 6.1**

<img src="SymPyBilleder/2025-11-12-19-06-26.png" width="550">

<img src="SymPyBilleder/2025-11-12-19-06-47.png" width="550">

<img src="SymPyBilleder/2025-11-12-19-07-07.png" width="550">

**Answer:**

We know something in the book has the formula:
$$
D(vc(Y)) = \Sigma \otimes I_n
$$
Here, we have $n=2$ observations so:
$$
= \Sigma \otimes I_2
$$
$$
= \begin{bmatrix} \sigma I_2 & \gamma I_2 & \rho I_2 \\ \gamma I_2 & \sigma I_2 & \gamma I_2 \\ \rho I_2 & \gamma I_2 & \sigma I_2 \end{bmatrix}
$$
Here, $I_2$ is the identity matrix of size $2 \times 2$, so like for instance we must have a $2 \times 2 $ matrix with only $\sigma$ on the diagonal, and the same with $\gamma$ and $\rho$ and so on...

The only option that matches is the last one.

**The answer is, therefore, option 6.**

### **Question 7.1**

<img src="SymPyBilleder/2025-11-12-19-07-43.png" width="550">

<img src="SymPyBilleder/2025-11-12-19-07-53.png" width="550">

**Answer:**

First, canonical correlation is $r$ and the quesiton asks for the variance which is $r^2$, so we just take the first variable's canonical correlation and square it.
$$
0.3083^2 = 0.0950
$$
**The answer is, therefore, option 6.**

### **Question 7.2**

<img src="SymPyBilleder/2025-11-12-19-08-09.png" width="550">

**Answer:**

For the first data set, there is $6$ variables, and for the second there is $11$, and there is $528$ observations in both datasets, so we use:
$$
U(p, q, n - p - q -1)
$$
SO:
$$
U(6, 11, 528 - 11 - 1) = U(6, 11, 516)
$$
**The answer is, therefore, option 2.**

FROM remark 6.15 asking "we may test whether $Y$ and $X$ are independent by testing whether the first canonical correlation is $0$"...

### **Question 7.3**

<img src="SymPyBilleder/2025-11-12-19-08-23.png" width="550">

**Answer:**

It should be $12$, because:
$$
\Sigma_{XX} \text{is full rank (invertible)} \Leftrightarrow n>p
$$
So aka we just have to do $p+1 = 11 + 1 = 12$.

**The answer is, therefore, option 6.**